sdk/identity/Azure.Identity/src/ManagedIdentityClient.cs

```diff
@@ -18,8 +18,8 @@ internal class ManagedIdentityClient
     {
         private const string AuthenticationResponseInvalidFormatError = "Invalid response, the authentication response was not in the expected format.";
         private const string MsiEndpointInvalidUriError = "The environment variable MSI_ENDPOINT contains an invalid Uri.";
-        internal const string MsiUnavailableError = "No managed identity endpoint found.";
-        internal const string IdentityUnavailableError = "The requested identity has not been assigned to this resource.";
+        internal const string MsiUnavailableError = "ManagedIdentityCredential not available, no managed identity endpoint found.";
```

Comment:
Just wondering if you intended to also prefix the two errors above with "ManagedIdentityCredential not available, " as well.

I don't have a strong opinion either way.

Reply:
See my suggestion above to reduce words per credential error

Reply:
No the top two errors are actually fatal errors not just errors which result in a CredentialUnavailableException.